src/util/ringdelaybuffer.h

```diff
@@ -0,0 +1,142 @@
+/*
+* $Id$
+* Portable Audio I/O Library
+* Ring Buffer utility.
+*
+* Author: Phil Burk, http://www.softsynth.com
+* modified for SMP safety on OS X by Bjorn Roche.
+* also allowed for const where possible.
+* modified for multiple-byte-sized data elements by Sven Fischer
+*
+* Note that this is safe only for a single-thread reader
+* and a single-thread writer.
```

Comment:
```suggestion
```
instead add explanation down below that this is actually only intended to be single-threaded.